SabkTrial/Scenes/News Screen/NewsScreenModel.swift

```diff
@@ -7,42 +7,51 @@
 //
 
 import Foundation
+import DateToolsSwift
 
 class NewsScreenModel: BaseModel, NewsScreenModelProtocol {
- 
-    private var materials : [Materials] = []
-    private var sliders : [Slider] = []
-    private var images : [Comics] = []
-    private var videos : [Comics] = []
-    private var articles : [Materials] = []
-
-    func fetchNewsData(compelation: @escaping (Bool) -> Void){
+    
+    private var materials: [Materials] = []
+    private var sliders: [Slider] = []
+    private var images: [Comics] = []
+    private var videos: [Comics] = []
+    private var articles: [Materials] = []
+    
+    func fetchNewsData(compelation: @escaping (Bool) -> Void) {
         getNews(compelation: {result in
-            switch result{
+            switch result {
             case .success(let response):
                 let data = response as? SliderMaterialResponse
-                if let dataReturned = data{
-                    if let slid = dataReturned.slider{
+                if let dataReturned = data {
+                    if let slid = dataReturned.slider {
                         self.sliders = slid
-                    }else {
+                        for index in 0..<self.sliders.count {
+                            self.sliders[index].timeApart =
+                                self.getTimeApart(forPublishDate: self.sliders[index].publishDate ?? "")
+                        }
+                    } else {
                         print("ERROR : sliders arent available")
                     }
-                    if let material = dataReturned.materials{
+                    if let material = dataReturned.materials {
                         self.materials = material
-                        if(!self.videos.isEmpty){
+                        for index in 0..<self.materials.count {
+                            self.materials[index].timeApart =
+                                self.getTimeApart(forPublishDate: self.materials[index].publishDate ?? "")
+                        }
+                        if(!self.videos.isEmpty) {
                             self.materials.insert(Materials(type: Materialtypes.videos), at: 4)
                         }
-                        if(!self.images.isEmpty){
+                        if(!self.images.isEmpty) {
                             self.materials.insert(Materials(type: Materialtypes.images), at: 9)
                         }
```

Comment:
there is a better look than this
 this class need to be refactored

Reply:
done